utils/filter/filter.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 Red Hat, Inc.
+// All Rights Reserved.
+//
+//    Licensed under the Apache License, Version 2.0 (the "License"); you may
+//    not use this file except in compliance with the License. You may obtain
+//    a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+package filter
+
+import (
+	"strings"
+
+	"github.com/redhatcre/syncron/pkg/cli"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+func Component(filter []string) []string {
+
+	var filtered []string
+	test := viper.GetStringSlice(cli.SOSReports)
+	for _, file := range test {
```

Comment:
I have a couple questions about this nested loop:
1) Can a file be matched to more than one component?
2) Are the filtered files supposed to be unique? If files can match more than one component, then it could be problematic (the `filtered` slice would contain the same file multiple times)

Reply:
This is a very good point @cescgina , because right now if it matches x times, it will try to download the file x times.
I will fix it, thank you.